README.md

```diff
@@ -14,18 +14,55 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 # Report
 
 ## Specify if you have used a linear time or quadratic time algorithm.
+We ended up with a quadratic time algorithm, we played around with mccreight, but other projects got the better of us, and ate what little time we had left.
 
 ## Insights you may have had while implementing and comparing the algorithms.
+Our implementation choice on how to handle leaves (in this case they're just ints) sparked a lengthy yet very enlightening discussion on how we interpret and envision abstract data structure in our minds eye, turns out we each have our differences.  
+
+Packing all our functions into the classes might not be the smartes way of doing this (having the bft as a method under SuffixTree make the usage of the method a bit unintuitive.), but it made importing nice and easy.
 
 ## Problems encountered if any.
+Other than getting one step closer to enlightenment, most problems we encountered were -as usual- temporary strokes of idiocy on our part (indexing -once again- is hard).
+Most difficulties we had were in early implemetation choices for our Knæ class, eg. do we want a dictionary or list implementation, and how do we handled it. How much information do we need in each Knæ, and how do we deal with our choice of ints as leaves. 
 
 ## Correctness
 
 *Describe experiments that verifies the correctness of your implementations.*
 
+We could not come up with a good idea for testing just the construction, but if the tree is constructed correctly the search should output the same matches as the naive implementation of exact pattern matching. 
```

Comment:
There are a ton of properties a suffix tree must have. Testing some of those could be an idea...